cmd/nknd/commands/root.go

```diff
@@ -43,94 +41,66 @@ import (
+			log.Error(err)
+		}
+	},
+}
+
 var (
 	createMode bool
 )
 
-func init() {
-	runtime.GOMAXPROCS(runtime.NumCPU())
-	rand.Seed(time.Now().UnixNano())
-}
-
-func InitLedger(account *vault.Account) error {
-	var err error
-	store, err := store.NewLedgerStore()
-	if err != nil {
-		return err
-	}
-	blockChain, err := chain.NewBlockchainWithGenesisBlock(store)
+// Execute adds all child commands to the root command and sets flags appropriately.
+// This is called by main.main(). It only needs to happen once to the rootCmd.
+func Execute() {
+	err := rootCmd.Execute()
 	if err != nil {
-		return err
-	}
-	chain.DefaultLedger = &chain.Ledger{
-		Blockchain: blockChain,
-		Store:      store,
-	}
-
-	return nil
-}
-
-func printMemStats() {
-	var m runtime.MemStats
-	runtime.ReadMemStats(&m)
-	log.Infof("Alloc = %v TotalAlloc = %v Sys = %v NumGC = %v\n", m.Alloc/1024, m.TotalAlloc/1024, m.Sys/1024, m.NumGC)
-	log.Infof("HeapAlloc = %v HeapSys = %v HeapIdle = %v HeapInuse = %v HeapReleased = %v HeapObjects = %v\n", m.HeapAlloc/1024, m.HeapSys/1024, m.HeapIdle/1024, m.HeapInuse/1024, m.HeapReleased/1024, m.HeapObjects/1024)
-	log.Infof("StackInuse = %v StackSys = %v MCacheInuse = %v MCacheSys = %v\n", m.StackInuse/1024, m.StackSys/1024, m.MCacheInuse/1024, m.MCacheSys/1024)
-}
-
-func JoinNet(nn *nnet.NNet) error {
-	seeds := config.Parameters.SeedList
-	rand.Shuffle(len(seeds), func(i int, j int) {
-		seeds[i], seeds[j] = seeds[j], seeds[i]
-	})
-
-	for _, seed := range seeds {
-		randAddrs, err := client.FindSuccessorAddrs(seed, util.RandomBytes(config.NodeIDBytes))
-		if err != nil {
-			log.Warningf("Can't get successor address from [%s]", seed)
-			continue
-		}
-
-		for _, randAddr := range randAddrs {
-			if randAddr == nn.GetLocalNode().Addr {
-				log.Warning("Skipping self...")
-				continue
-			}
-			err = nn.Join(randAddr)
-			if err != nil {
-				log.Error(err)
-				continue
-			}
-			return nil
-		}
+		os.Exit(1)
 	}
-	return errors.New("Failed to join the network")
 }
 
-// AskMyIP request to seeds randomly, in order to obtain self's externIP and corresponding chordID
-func AskMyIP(seeds []string) (string, error) {
-	rand.Shuffle(len(seeds), func(i int, j int) {
-		seeds[i], seeds[j] = seeds[j], seeds[i]
-	})
+func init() {
+	runtime.GOMAXPROCS(runtime.NumCPU())
+	rand.Seed(time.Now().UnixNano())
 
-	for _, seed := range seeds {
-		addr, err := client.GetMyExtIP(seed, []byte{})
-		if err == nil {
-			return addr, err
-		}
-		log.Warningf("Ask my ID from %s error: %v", seed, err)
-	}
-	return "", errors.New("Tried all seeds but can't got my external IP and nknID")
+	rootCmd.CompletionOptions.DisableDefaultCmd = true
+	rootCmd.Flags().BoolVarP(&createMode, "create", "c", false, "Create Mode")
+	rootCmd.Flags().StringVar(&config.SeedList, "seed", "", "Seed node address to join, multiple seeds should be split by comma")
+	rootCmd.Flags().StringVarP(&password.Passwd, "passwd", "p", "", "Password of Your wallet private Key")
```

Comment:
Deprecated -p option for password since it should be showed in `ps` or others process info.